src/crypto/secrets_manager/secrets_interface.rs

```diff
@@ -0,0 +1,25 @@
+//! Secrets management interface
+
+use masking::Secret;
+
+/// Trait defining the interface for managing application secrets
+#[async_trait::async_trait]
+pub trait SecretManager: Send + Sync {
+    /// Given an input, decrypt/retrieve the secret
+    async fn get_secret(
+        &self,
+        input: Secret<String>,
+    ) -> error_stack::Result<Secret<String>, SecretsManagementError>;
```

Comment:
So, here we are considering that the return type and the input will always be a `Secret<String>` but what if in case when this isnt' true?

Reply:
in what case?

Reply:
In the future, might we have a usecase where we need to return let's say `Vec<u8>`?